spark/dl/src/main/scala/com/intel/analytics/bigdl/utils/DistriParameterSynchronizer.scala

```diff
@@ -199,249 +170,347 @@ class BlockManagerParameterSynchronizer[T: ClassTag](partitionID: Int,
     })
   })
 
-  private val syncMetaMap = new ConcurrentHashMap[String, SyncMeta[T]]
-
-  override def init(name: String, globalSize: Int, priority: Int = 1): Unit = {
-    val partitionToCount = if (globalSize < totalPartition) globalSize else totalPartition
-    syncMetaMap.putIfAbsent(name, SyncMeta(name, 1, priority, globalSize, partitionToCount,
-      new ConcurrentHashMap[Int, CompressedTensor[T]](),
-      new ConcurrentHashMap[Int, Tensor[T]]()))
-  }
-
-  override def put(name: String, parameter: Tensor[T]): Unit = {
-    val syncMeta = syncMetaMap.get(name)
-    val asyncTask = new AsyncTask(syncMeta, parameter)
-    val futureTask = new FutureTask[Tensor[T]](asyncTask)
-    val futureAsyncTask = new AsyncFutureTask(futureTask, syncMeta.priority)
-    asyncTaskWaitingQueue.add(futureAsyncTask)
-    val clearTask = new ClearTask(name, syncMeta.counter - 1,
-      partitionID, syncMeta.partitionToCount)
-    clearPool.execute(clearTask)
-    syncResults.put(name, futureTask)
-  }
-
-  override def get(name: String): Tensor[T] = {
-    require(syncResults.contains(name), "put must be done before get")
-    val res = syncResults.get(name).get.get()
-    val syncMeta = syncMetaMap.get(name)
-    syncMeta.counter += 1
-    res
-  }
-
-  private class ClearTask(name: String, counter: Int, partitionID: Int,
-                         partitionToCount: Int)
-    extends Runnable {
-    override def run(): Unit = {
-      (0 until partitionToCount).foreach(pid => {
-        val parameterBlockId = getParameterBlockId(name,
-          counter, partitionID, pid)
-        BlockManagerWrapper.removeBlock(parameterBlockId)
-      })
-      // only local parititon < partitionToCount, there are aggregated blocks
-      if (partitionID < partitionToCount) {
-        val aggregatedParameterBlockId = getParameterBlockId(s"${name}_aggregated",
-          counter, partitionID, -1)
+  // to process request
+  private val workerPool: ExecutorService =
+    Executors.newFixedThreadPool(workerPoolSize, new ThreadFactory {
+      override def newThread(r: Runnable): Thread = {
+        val t = Executors.defaultThreadFactory().newThread(r)
+        t.setDaemon(true)
+        t
```

Comment:
Should we define a class to maintain this our implementation of `ThreadFactory` ?

Reply:
Just keep the original `Utils` style